crd/hack/update-codegen.sh

```diff
@@ -48,39 +48,51 @@ if [[ "${VERIFY_CODEGEN:-}" == "true" ]]; then
 fi
 
 readonly COMMON_FLAGS="${VERIFY_FLAG:-} --go-header-file ${SCRIPT_ROOT}/hack/boilerplate.go.txt"
-readonly PLURAL_EXCEPTIONS="--plural-exceptions=GKENetworkParams:GKENetworkParams"
 
 generate_config() {
-  local crd_name version apis_pkg
-  crd_name=${1}
-  version=${2}
-  apis_pkg=${APIS_BASE_PKG}/${1}/${2}
+  local crd_name version crd_version apis_pkg
+  crd_name="$1"
+  version="$2"
+  crd_version="$3"
+  apis_pkg="${APIS_BASE_PKG}/${1}/${2}/..."
+
+  if [[ $# == 4 ]]; then
+    apis_pkg="{${apis_pkg},${APIS_BASE_PKG}/${1}/${4}/...}"
+  fi
 
   echo "Performing code generation for ${crd_name} CRD"
   echo "Generating deepcopy functions and CRD artifacts"
   go run sigs.k8s.io/controller-tools/cmd/controller-gen \
-          object:headerFile=${SCRIPT_ROOT}/hack/boilerplate.go.txt \
-          crd:crdVersions=$version \
-          paths=${apis_pkg}/... \
-          output:crd:artifacts:config=${SCRIPT_ROOT}/config/crds
+          object:headerFile="${SCRIPT_ROOT}/hack/boilerplate.go.txt" \
+          crd:crdVersions="${crd_version}" \
+          paths="${apis_pkg}" \
+          output:crd:artifacts:config="${SCRIPT_ROOT}/config/crds"
 }
 
 codegen_for () {
-  local crd_name version apis_pkg output_pkg
+  local crd_name version crd_version extra_version apis_pkg output_pkg
 
-  if [[ $# != 2 ]]; then
-    echo "Usage: codegen_for CRD-NAME VERSION" >&2
+  if [[ $# != 3 ]] && [[ $# != 4 ]]; then
```

Comment:
you don't have to change this, but technically "=" is string comparison

Reply:
changed